TypeCobol.LanguageServer/Completion Factory/CompletionFactoryHelpers.cs

```diff
@@ -224,5 +230,82 @@ public static CompletionItem CreateCompletionItemForIndex(Compiler.CodeElements.
             var display = string.Format("{0} (from {1})", index.Name, variable.Name);
             return new CompletionItem(display) {insertText = index.Name, kind = CompletionItemKind.Variable};
         }
+
+        public static Case GetTextCase(string tokenText)
+        {
+            if (string.IsNullOrEmpty(tokenText)) return Case.Lower;
+            // check if upper case
+            bool isUpper = true;
+            foreach (char c in tokenText)
+            {
+                if (char.IsLower(c))
+                {
+                    isUpper = false;
+                    break;
+                }
+            }
+
+            if (isUpper) return Case.Upper;
+            // check if Camel case; ex. CamlSample
+            bool isCamel = false;
+            int iUpperMem = -1, count = tokenText.Length;
+            for (int i = 0; i < count; i++)
+            {
+                if (i == 0)
+                {
+                    // first letter
+                    if (char.IsLower(tokenText[i])) break;
+                    isCamel = true;
+                }
+                else
+                {
+                    if (char.IsUpper(tokenText[i]))
+                    {
+                        if (iUpperMem == (i - 1))
+                        {
+                            // 2 upper case characters one behind the other
+                            isCamel = false;
+                            break;
+                        }
+                        iUpperMem = i;
+                    }
+                }
+            }
+
+            if (isCamel && iUpperMem == count - 1) isCamel = false; // last char is upper
+            return (isCamel) ? Case.Camel : Case.Lower;
+        }
+
+        public const string ParamInput = "INPUT";
+        public const string ParamOutput = "OUTPUT";
+        public const string ParamInOut = "IN-OUT";
+        
+        public static Dictionary<string, string> GetParamsWithCase(Case textCase)
+        {
+            if (textCase == Case.Upper) return new Dictionary<string, string>()
+            {
+                { ParamInput, "INPUT"},
+                { ParamOutput, "OUTPUT" },
+                { ParamInOut, "IN-OUT" }
+            };
+            if (textCase == Case.Lower) return new Dictionary<string, string>()
+            {
+                { ParamInput, "input"},
+                { ParamOutput, "output" },
+                { ParamInOut, "in-out" }
+            };
+            return new Dictionary<string, string>()     // camel case
+            {
+                { ParamInput, "Input"},
+                { ParamOutput, "Output" },
+                { ParamInOut, "In-Out" }
+            };
+        }
+        public enum Case
+        {
+            Upper,
+            Camel,
+            Lower       // default value
```

Comment:
done
test LSR "ProcedureCompletionInputInoutOutput-Case"
https://github.com/TypeCobolTeam/TypeCobol/tree/6d2502c2ac02e718a765ac3e549933b59b6ebb69

> Yes I wait demonstration of the tool LanguageServerRobot with Jean, friday De : Olivier Smedile [mailto:notifications@github.com] Envoyé : jeudi 17 octobre 2019 17:02 À : TypeCobolTeam/TypeCobol <TypeCobol@noreply.github.com> Cc : LANOY Dominique (PRESTA EXT) <dominique.lanoy@externe.e-i.com>; Assign <assign@noreply.github.com> Objet : Re: [TypeCobolTeam/TypeCobol] WI #1378 Manage completion about call and its keywords by considering… (#1516) @smedilol requested changes on this pull request. Can you create a LSR test for this issue ? — You are receiving this because you were assigned. Reply to this email directly, view it on GitHub<#1516?email_source=notifications&email_token=ANFBGXVM7F4DXJ7VLZK5LNTQPB47PA5CNFSM4JB2DHL2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCIKGUHI#pullrequestreview-303327773>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ANFBGXU6HMONUEGPCDRU7D3QPB47PANCNFSM4JB2DHLQ>. *********************************************************************************************** Ce message et toutes les pièces jointes sont confidentiels et établis à l'intention exclusive de son ou ses destinataires. Si vous avez reçu ce message par erreur, merci d'en avertir immédiatement l'émetteur et de détruire le message. Toute modification, édition, utilisation ou diffusion non autorisée est interdite. L'émetteur décline toute responsabilité au titre de ce message s'il a été modifié, déformé, falsifié, infecté par un virus ou encore édité ou diffusé sans autorisation. *********************************************************************************************** This message and any attachments are confidential and intended for the named addressee(s) only. If you have received this message in error, please notify immediately the sender, then delete the message. Any unauthorized modification, edition, use or dissemination is prohibited. The sender shall not be liable for this message if it has been modified, altered, falsified, infected by a virus or even edited or disseminated without authorization. ***********************************************************************************************

https://github.com/TypeCobolTeam/TypeCobol/tree/6d2502c2ac02e718a765ac3e549933b59b6ebb69